src/ast/inline-document-descriptor.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+export class InlineDocumentDescriptor {
+  type: 'html' | 'javascript' | 'css' | string;
```

Comment:
Minor improvement maybe:

```typescript
type: 'html' | 'javascript' | 'css' | /* etc... */string;
```

I'm big on selling typescript for our code as machine-checked documentation.